pkg/sql/exec/float.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package exec
+
+import "math"
+
+// compareFloats compares two float values. This function is necessary for NaN
+// handling. In SQL, NaN is treated as less than all other float values. In Go,
+// any comparison with NaN returns false.
+func compareFloats(a, b float64) int {
+	if a < b {
+		return -1
+	}
+	if a == b {
+		return 0
+	}
+	if a > b {
+		return 1
+	}
+	if math.IsNaN(a) {
+		if math.IsNaN(b) {
```

Comment:
I wonder if we could avoid this second check by changing `a == b` to be `int64(a) == int64(b)` or whatever - aren't the bits of NaN the same?

Reply:
Good thought. I think we would use `math.Float64bits`.

Reply:
BTW, I was worried this might be slower than the normal float equality check, but it's actually faster! Presumably because the normal one has to do NaN checks.